### PR TITLE
Clear the session-user cache when changing _User objects

### PR DESCRIPTION
**Pull request by gfosco**
For issue #1062

### CODE REVIEW

File: src/RestWrite.js [hunk 1]
```diff
@@ -341,6 +341,11 @@ RestWrite.prototype.transformUser = function() {
     });
   }
 
+  // If we're updating a _User object, clear the user cache for the session
```
Comment (flovilmart) [inline src/RestWrite.js]: Also when we have the oauthLogin shortcut?

**Comment by mullwaden** [comment]
Thanks @gfosco !

Does it also solve for the following case where we have a user with a pointer to a second object (privateData)?
* some change is made to the user such that the privateData is included in the cache (this seems to happen)
* I make a change to the privateData object (without touching the user)
* I create a new object card details (on which I use  cardDetails.set('user', request.user)

Will there be an old version of the "privateData" object tagging along with the cardDetails?

File: src/RestWrite.js (hunk 1 above)
Comment (gfosco) [inline src/RestWrite.js]: Not sure what you mean here..

**Comment by gfosco** [comment]
I don't think it would solve that case, no.  I think you should perform a fetch on any included objects.

Since we store the user cache based on sessionToken, it also wouldn't support changes to that user made by the masterKey.  We'd have to search through all the sessions to invalidate any matching that user...

File: src/RestWrite.js (hunk 1 above)
Comment (flovilmart) [inline src/RestWrite.js]: I mean when we do the login shortcut with oAuth, we go through that method. If there is a cached token for the found user, we should destroy it too to remove any wrong possible state, the same way we remove it upon user login

**Pull request merged**